front/src/applications/operationalStudies/hooks/usePathProjection.ts

```diff
@@ -1,21 +1,71 @@
-import { useMemo } from 'react';
+import { useEffect, useMemo } from 'react';
 
 import { skipToken } from '@reduxjs/toolkit/query/react';
+import type { TFunction } from 'i18next';
 import { isEqual } from 'lodash';
+import { useTranslation } from 'react-i18next';
 import { useSelector } from 'react-redux';
 
-import { osrdEditoastApi, type PathfindingResult } from 'common/api/osrdEditoastApi';
+import {
+  osrdEditoastApi,
+  type PathfindingResult,
+  type OperationalPointReference,
+  type PathProperties,
+} from 'common/api/osrdEditoastApi';
 import { getExceptionFromOccurrenceId } from 'modules/timetableItem/helpers/pacedTrain';
 import type { TimetableItemId, TimetableItem } from 'reducers/osrdconf/types';
-import { getTrainIdUsedForProjection } from 'reducers/simulationResults/selectors';
+import { updateProjectionType } from 'reducers/simulationResults';
+import {
+  getProjectionType,
+  getTrainIdUsedForProjection,
+} from 'reducers/simulationResults/selectors';
+import { useAppDispatch } from 'store';
+import { formatUicToCi } from 'utils/strings';
 import {
   extractEditoastIdFromPacedTrainId,
   extractEditoastIdFromTrainScheduleId,
   extractPacedTrainIdFromOccurrenceId,
+  isOccurrenceId,
   isPacedTrainId,
   isTrainScheduleId,
 } from 'utils/trainId';
 
+import type { PathProjectionResult } from '../types';
+import { getStationFromOps, isOperationalPointReference } from '../utils';
+
+/**
+ * Generates a display name for a virtual operational point based on available reference data.
+ * Uses trigram --> UIC --> operational_point --> position-based fallbacks in priority order.
+ */
+const getVirtualOpName = (
+  opRef: OperationalPointReference,
+  index: number,
+  totalCount: number,
+  t: TFunction<'operational-studies'>
+): string => {
+  if ('trigram' in opRef && opRef.trigram) {
+    return opRef.trigram;
+  }
+
+  if ('uic' in opRef && opRef.uic) {
+    return opRef.uic.toString();
+  }
+
+  if ('operational_point' in opRef && opRef.operational_point) {
+    return opRef.operational_point;
+  }
+
+  if (index === 0) {
+    return t('main.requestedOrigin');
+  }
+
+  if (index === totalCount - 1) {
+    return t('main.requestedDestination');
+  }
+
+  return t('main.requestedPoint', { count: index + 1 });
```

Comment:
nit: you can make this function more compact & readable like this
```suggestion
  if ('trigram' in opRef && opRef.trigram) return opRef.trigram;
  if ('uic' in opRef && opRef.uic) return opRef.uic.toString();
  if ('operational_point' in opRef && opRef.operational_point) return opRef.operational_point;

  if (index === 0) return t('main.requestedOrigin');
  if (index === totalCount - 1) return t('main.requestedDestination');
  return t('main.requestedPoint', { count: index + 1 });
```